apm-sniffer/apm-sdk-plugin/spymemcached-2.x-plugin/src/test/java/org/skywalking/apm/plugin/spymemcached/v2/MemcachedMethodInterceptorTest.java

```diff
@@ -0,0 +1,93 @@
+package org.skywalking.apm.plugin.spymemcached.v2;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.when;
+
+import java.lang.reflect.Method;
+import java.util.List;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.modules.junit4.PowerMockRunnerDelegate;
+import org.skywalking.apm.agent.core.context.trace.AbstractTracingSpan;
+import org.skywalking.apm.agent.core.context.trace.SpanLayer;
+import org.skywalking.apm.agent.core.context.trace.TraceSegment;
+import org.skywalking.apm.agent.core.context.util.KeyValuePair;
+import org.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.skywalking.apm.agent.test.helper.SegmentHelper;
+import org.skywalking.apm.agent.test.helper.SpanHelper;
+import org.skywalking.apm.agent.test.tools.AgentServiceRule;
+import org.skywalking.apm.agent.test.tools.SegmentStorage;
+import org.skywalking.apm.agent.test.tools.SegmentStoragePoint;
+import org.skywalking.apm.agent.test.tools.TracingSegmentRunner;
+
+import net.spy.memcached.MemcachedClient;
+
+@RunWith(PowerMockRunner.class)
+@PowerMockRunnerDelegate(TracingSegmentRunner.class)
+public class MemcachedMethodInterceptorTest {
+
+	 	@SegmentStoragePoint
+	    private SegmentStorage segmentStorage;
+	    @Rule
+	    public AgentServiceRule serviceRule = new AgentServiceRule();
+	    @Mock
+	    private EnhancedInstance enhancedInstance;
+	    private MemcachedMethodInterceptor interceptor;
+
+	    private Object[] allArgument;
+	    private Class[] argumentType;
+	    
+	    @Before
+	    public void setUp() throws Exception {
+	        allArgument = new Object[] {"OperationKey", "OperationValue"};
+	        argumentType = new Class[] {String.class, String.class};
+
+	        interceptor = new MemcachedMethodInterceptor();
+	        when(enhancedInstance.getSkyWalkingDynamicField()).thenReturn("127.0.0.1:11211");
+	    }
+
+	    @Test
+	    public void testIntercept() throws Throwable {
+	        interceptor.beforeMethod(enhancedInstance, getMockSetMethod(), allArgument, argumentType, null);
```

Comment:
You should test the future with exception scenario

Reply:
feature, not future, right? @ascrutae